lib/index.js

```diff
@@ -12,34 +26,307 @@ class PluginPassportOAuth {
+        }
+      }
+    })
+      .then(result => {
+        if (result.total === 0) {
+          return Promise.resolve(null);
+        }
+
+        return Promise.resolve(result.hits[0]);
+      });
+  }
+
+  /**
+   * Verify function called by passport to log the user.
+   * If a persist option is in the configuration of the strategy then every field present in this configuration will be persisted in the repository.
+   *
+   * @param request
+   * @param accessToken
+   * @param refreshToken
+   * @param profile
+   * @returns {Promise.<T>}
+   */
+  verify (request, accessToken, refreshToken, profile) {
+    const
+      user = {};
+
+    return this.getCredentialsFromUserId(profile._json[this.config.strategies[profile.provider].identifierAttribute], profile.provider)
+      .then(userObject => {
+        let req = null;
+
+        if (userObject !== null) {
+          return Promise.resolve(userObject.userId);
+        }
+
+        Object.keys(profile._json).forEach(attr => {
+          if (this.config.strategies[profile.provider].persist.indexOf(attr) > -1) {
+            user[attr] = profile._json[attr];
+          }
+        });
+
+        if (this.config.strategies[profile.provider].persist.length !== 0) {
+          req = {
+            controller: 'security',
+            action: 'createUser',
+            _id: profile._json[this.config.strategies[profile.provider].identifierAttribute],
+            body: {
+              content: {
+                profileIds: this.config.defaultProfiles ? this.config.defaultProfiles : ['default']
+              }, // content will be persisted in Kuzzle
+              credentials: {} // credentials will be persisted in the repository
+            }
+          };
+
+          if (this.config.strategies[profile.provider].kuzzleAttributesMapping) {
+            Object.keys(this.config.strategies[profile.provider].kuzzleAttributesMapping)
+              .forEach(attr => {
+                req.body.content[attr] = profile._json[this.config.strategies[profile.provider].kuzzleAttributesMapping[attr]];
+              });
+          }
+
+          user._id = profile._json[this.config.strategies[profile.provider].identifierAttribute];
+          req.body.credentials[profile.provider] = user;
+
+          return this.context.accessors.execute(new this.context.constructors.Request(request.original, req))
+            .then(() => {
+              return Promise.resolve(profile._json[this.config.strategies[profile.provider].identifierAttribute]);
+            });
+        }
+        return Promise.resolve(profile._json[this.config.strategies[profile.provider].identifierAttribute]);
+      })
+      .catch(err => Promise.reject(err));
+  }
+
+  /**
+   * Check if a given user exists
+   *
+   * @param request
+   * @param userId
+   * @param provider
+   * @returns {Promise|Promise.<boolean>}
+   */
+  exists (request, userId, provider) {
+    return this.getCredentialsFromUserId(userId, provider)
```

Comment:
`userId` here is the user id in Kuzzle, so we should just call this.getProviderRepository(provider).get(userId)